topography/topography.F90

```diff
@@ -244,27 +248,29 @@ function get_topog_mean_2d (blon, blat, zmean, use_mpp_io_arg)
    real, intent(out), dimension(:,:) :: zmean
    logical :: get_topog_mean_2d
    logical,optional :: use_mpp_io_arg
-
+   logical          :: mpp_io_use
 !-----------------------------------------------------------------------
-   if ( use_mpp_io .OR. present(use_mpp_io_arg) .AND. use_mpp_io_arg) then
-       get_topog_mean_2d = get_topog_mean_2d_mpp_io (blon, blat, zmean)
-       return
-   endif !< use_mpp_io
-
    if (.not. module_is_initialized) call topography_init()
 
+   if(present(use_mpp_io_arg)) then
+     mpp_io_use = use_mpp_io_arg
+   else
+     mpp_io_use = use_mpp_io
+   endif
+
    if ( any(shape(zmean(:,:)) /= (/size(blon,1)-1,size(blon,2)-1/)) .or. &
         any(shape(zmean(:,:)) /= (/size(blat,1)-1,size(blat,2)-1/)) ) &
         call error_mesg('get_topog_mean_2d','shape(zmean) is not&
             & equal to (/size(blon,1)-1,size(blon,2)-1/))', FATAL)
 
-   if ( open_topog_file() ) then
-       call interp_topog_2d ( blon, blat, zmean )
-       get_topog_mean_2d = .true.
+   if( mpp_io_use ) then
+     get_topog_mean_2d = open_topog_file_mpp_io(topog_file)
+     if ( get_topog_mean_2d ) call interp_topog_2d ( blon, blat, zmean,&
+                        use_mpp_io_arg=.true. )
    else
-       get_topog_mean_2d = .false.
+     get_topog_mean_2d = open_topog_file_mpp_io(topog_file)
```

Comment:
We are in the else, but still referencing an mpp_io function.

Reply:
Also fixed this call in [3ca8114](https://github.com/NOAA-GFDL/FMS/pull/571/commits/3ca81142d4dfa766fc9f9dca73b05211cef43acb)